FirebaseSharedSwift/Tests/third_party/DataEncoderTests.swift

```diff
@@ -482,6 +482,35 @@ class TestFirebaseDataEncoder: XCTestCase {
                    dataDecodingStrategy: .custom(decode))
   }
 
+  func testDecodingBase64StringAsBlobData() {
+    let data = "abcdef".data(using: .utf8)!
+    let base64String = "YWJjZGVm"
+
+    let encoder = FirebaseDataEncoder()
+    encoder.dataEncodingStrategy = .base64
+    var payload: Any! = nil
+    do {
+      payload = try encoder.encode(data)
+    } catch {
+      XCTFail("Failed to encode \(Data.self): \(error)")
+    }
```

Comment:
Just throwing this out for there for future work... in case it was considered, another approach that removes the need for the `do {} catch { XCTFail(...) }` pattern is to mark the test method as `throws`. You wouldn't have the formatted error message in the failure but the throwing line should still appear in the logs– which should provide enough context as to what went wrong.

Reply:
Good idea, I switched to that approach in PR #10638.